torchrl/objectives/cql.py

```diff
@@ -493,36 +501,62 @@ def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
         else:
             tensordict_reshape = tensordict
 
-        device = self.device
-        td_device = tensordict_reshape.to(device)
+        td_device = tensordict_reshape.to(tensordict.device)
 
-        loss_qvalue, loss_alpha_prime, priority = self._loss_qvalue_v(td_device)
-        loss_actor = self._loss_actor(td_device)
-        loss_alpha = self._loss_alpha(td_device)
-        tensordict_reshape.set(self.tensor_keys.priority, priority)
-        if loss_actor.shape != loss_qvalue.shape:
-            raise RuntimeError(
-                f"Losses shape mismatch: {loss_actor.shape} and {loss_qvalue.shape}"
-            )
+        q_loss, metadata = self.q_loss(td_device)
+        cql_loss, cql_metadata = self.cql_loss(td_device)
+        if self.with_lagrange:
+            alpha_prime_loss, alpha_prime_metadata = self.alpha_prime_loss(td_device)
+            metadata.update(alpha_prime_metadata)
+        loss_actor_bc, bc_metadata = self.actor_bc_loss(td_device)
+        loss_actor, actor_metadata = self.actor_loss(td_device)
+        loss_alpha, alpha_metadata = self.alpha_loss(td_device)
+        metadata.update(
+            {
+                **bc_metadata,
+                **cql_metadata,
+                **actor_metadata,
+                **alpha_metadata,
+            }
```

Comment:
this assumes that all metadata are Dict[Str, Any] but maybe we'll add smth else?
Why not 4 updates? (at least this version will crash if 2 metadata collide)